### PR TITLE
383 Refactor JIRA tests

### LINKED ISSUE #383: Refactor JIRA Tests
### 1. Pupose
a. The unit tests currently in the ```tests/testthat/test-jira.R``` are not comprehensive, and do not cover all function categories that should be tested for. 

b. The existing JIRA tests only cover for ```parse_jira```, despite there being 3 downloaders, 3 parsers, and a refresher function to test for. 

### 2. Process
a. Add to the existing ```tests/testthat/test-jira.R``` file so that it tests all functions available. The code coverage progress and impact of my work can be visualized with codecov, which is a coverage tool that informs how much code is tested for in Kaiaulu's packages. Based on previous notebooks, I’ve updated my test table to contain three main types of tests. 
**Test Explanations**
Types of Test | Test Explanation |
| --- | --- |
| Path Test | Tests that the both the incorrect and correct arguments for a function either result in the function successfully running or correctly erroring |
| Behavioral Test | Tests that functions will behave correctly given certain inputs |
| Third Party Tools Test | Checks if a path for a third party tool being called has been updated in ```tools.yml```, and only test if the tool has been set up correctly |

b. Identify all tests in ```test-jira.R```, refine them, and then create tests of my own to fully test all functions in the file. While moving through the files, I will first create Behavioral tests for the code inside the functions, and then add the Path and Third Party Tools tests after gaining a better understanding of how the functions behave and interact with each other. I will then compile a full proposed test list with statements written in the format of Kaiaulu’s testing package ```testthat```.  A basic example of what this would look like is `”testthat if the verbose argument is set to TRUE, function will print a download message”`. These tests would be uploaded in the form of a .md file in a Pull Request that is linked to this issue. Once my test list is approved by @carlosparadis, I will begin work coding all of my tests and add them to the Pull Request to be merged into the main branch of Kaiaulu. 

**Test Process Checklist**
- [x] Identify all current JIRA tests
- [ ] Create Behavioral Tests for every function
- [ ] Create Path Tests for every function
- [ ] Create Third Party Tests(If applicable)
- [ ] Compile a full test list and create a Pull Request with it in a .md file
- [ ] Code the tests in R
- [ ] Update the Pull Request with the tests and pass CI/CD checks
- [ ] Waiting for Architect + Stakeholder review
- [ ] Merge the code

### PR DESCRIPTION
**Pull request by cfuke1**
This Pull Request contains the tests I wrote for the JIRA parser, along with the fake Example functions. Most of these tests are meant to test edge cases with the parser.

### CODE REVIEW

File: tests/testthat/test-jira.R [hunk 1]
```diff
@@ -39,3 +39,130 @@ test_that("parse_jira parses one issue with two comments as two rows", {
...
+  # The test should cause an error from parse_jira, but instead creates a data table that doesn't contain an issue_key.
+
+  jira_json_path <- example_jira_missing_key(folder_path = "/tmp", folder_name = "missing_issue_key")
+
+  expect_error(parse_jira(json_folder_path = jira_json_path))
+
+  io_delete_folder(folder_path="/tmp", folder_name="missing_issue_key")
+})
+
+test_that("When parse_jira is given a descrpition that contains an escape character, then it returns an unescaped result in the data table", {
```
Comment (Michelle4929) [inline tests/testthat/test-jira.R]: You have a typo here for "description", at the moment it says "descrpition".

File: R/example.R [hunk 2]
```diff
@@ -427,7 +427,212 @@ example_notebook_alternating_function_in_files <- function(folder_path, folder_n
...
+    fix_versions = c("1.1.1"),
+    assignee_name = "Moe",
+    creator_name = "Bob",
+    reporter_name = "Joe"
+  )
+
+  issues <- list(issue1)
+
+  jira_json_path <- make_jira_issue_tracker(issues,
+                                            save_filepath=file.path(folder_path, "ONE_ISSUE_NO_COMMMENTS_issues_1121646814_1121719175.json"))
```
Comment (Michelle4929) [inline R/example.R:467]: You have 3 M's in "COMMMENTS", is that correct?

File: tests/testthat/test-jira.R [hunk 3]
```diff
@@ -39,3 +39,130 @@ test_that("parse_jira parses one issue with two comments as two rows", {
   expect_equal(nrow(comments),2)
   }
 )
+
+test_that("When parse_jira is given a folder path containing jira issues as .json files, then it returns an issue and comments data table", {
+  jira_json_path <- example_jira_issue_components(folder_path = "/tmp", folder_name = "single_issue")
+  issues <- parse_jira(json_folder_path  = jira_json_path)
+
+  expect_is(issues$issue, "data.table")
```
Comment (Michelle4929) [inline tests/testthat/test-jira.R]: Just double checking but would this inconsistency cause an error for issues$issue? As every time you have issues$issues later down in the code there is an s on issue.

File: NEWS.md [hunk 4]
```diff
@@ -2,7 +2,7 @@ __kaiaulu 0.0.0.9700 (in development)__
 =========================
 
 ### NEW FEATURES
-
+ *  Added new tests and their example functions for ```parse_jira()``` in ```test-jira.R``` and ```R/Example.R``` respectively. [#383](https://github.com/sailuh/kaiaulu/issues/383)
```
Comment (john-a-flinn) [inline NEWS.md]: Other comments have 1 " ' "  not 3. It still looks the same so it probably doesn't matter

**Review by john-a-flinn — APPROVED** [review]
Minor typos, and strange formatting, no time to fix

**Review by john-a-flinn — CHANGES_REQUESTED** [review]
I missed @Michelle4929 comment. Does this code run?

File: tests/testthat/test-jira.R (hunk 3 above)
Comment (cfuke1) [reply tests/testthat/test-jira.R]: Good catch, the test was still running but I don't think it was comparing the correct tables.
